src/dataDisplay/table.jsx

```diff
@@ -1,21 +1,522 @@
-'use strict';
-
-import React, { Component } from 'react';
 import ClassNames from 'classnames';
+import {
+    debounce,
+    get,
+} from 'lodash';
 import PropTypes from 'prop-types';
+import React from 'react';
+import ScrollBar from 'react-custom-scrollbars';
+import { withStyles } from '../styles';
+import Utils from '../utils/utils';
 import TableBody from './tableBody';
 import TableCell from './tableCell';
+import {
+    DEBOUNCE_WAIT_TIME,
+    TH_RESIZABLE_MAX_WIDTH,
+} from './tableConstants';
 import TableHeader from './tableHeader';
-import TableHeaderCell from './tableHeaderCell';
 import TableRow from './tableRow';
+import TableHeaderCell from './tableHeaderCell';
+
+const propTypes = {
+    /**
+     * Basic (default) styling.
+     */
+    basic: PropTypes.bool,
+    /**
+     * If `true`, all columns will be divided.
+     */
+    celled: PropTypes.bool,
+    /**
+     * The content of the Table.
+     */
+    children: PropTypes.node,
+    /**
+     * Override or extend the styles applied to Table.
+     */
+    classes: PropTypes.shape({
+        tableSticky: PropTypes.string,
+        tableStickyColumns: PropTypes.string,
+    }),
+    /**
+     * Add additional classes to the Table.
+     */
+    className: PropTypes.string,
+    /**
+     * If `true`, applies `width: auto` to the Table.
+     */
+    collapsing: PropTypes.bool,
+    /**
+     * If `true`, far left column divided from the rest of the columns.
```

Comment:
Maybe "If \`true\` **_the_** far left column **_is_** divided from the rest of the columns."

Reply:
Fixed.

Reply:
Looks good; resolving.